spanner/api/Spanner.Samples.Tests/SpannerFixture.cs

```diff
@@ -69,51 +69,36 @@ private bool InitializeInstance()
         }
     }
 
-    private async Task DeleteStaleBackupsAndDatabasesAsync()
+
+    /// <summary>
+    /// Deletes first 5 database if no of database is more than 94.
+    /// </summary>
+    private async Task DeleteStaleDatabasesAsync()
     {
         DatabaseAdminClient databaseAdminClient = DatabaseAdminClient.Create();
         var instanceName = InstanceName.FromProjectInstance(ProjectId, InstanceId);
         var databases = databaseAdminClient.ListDatabases(instanceName)
             .Where(c => c.DatabaseName.DatabaseId.StartsWith("my-db-") || c.DatabaseName.DatabaseId.StartsWith("my-restore-db-"));
-        var databasesToDelete = new List<string>();
 
-        // Delete all the databases created before 48 hrs.
-        var timestamp = DateTimeOffset.UtcNow.AddHours(-48).ToUnixTimeMilliseconds();
-        foreach (var database in databases)
+        if (databases.Count() < 95)
         {
-            var databaseId = database.DatabaseName.DatabaseId.Replace("my-restore-db-", "").Replace("my-db-", "");
-            if (long.TryParse(databaseId, out long dbCreationTime) && dbCreationTime <= timestamp)
-            {
-                databasesToDelete.Add(database.DatabaseName.DatabaseId);
-            }
+            return;
         }
 
-        await Console.Out.WriteLineAsync($"{databasesToDelete.Count} old databases found.");
-
-        // Get backups.
-        ListBackupsRequest request = new ListBackupsRequest
-        {
-            ParentAsInstanceName = instanceName,
-            Filter = $"database:my-db-"
-        };
-        var backups = databaseAdminClient.ListBackups(request);
-
-        // Backups that belong to the databases to be deleted.
-        var backupsToDelete = backups.Where(c => databasesToDelete.Contains(DatabaseName.Parse(c.Database).DatabaseId));
-
-        await Console.Out.WriteLineAsync($"{backupsToDelete.Count()} old backups found.");
-
-        // Delete the backups.
-        foreach (var backup in backupsToDelete)
+        var databaseIdsWithCreatedTime = new Dictionary<string, long>();
```

Comment:
And I think you can substitute from here to line 99 with, so we don't have to iterate many tames and create a dictionary (can't add suggestion for deleted lines), note that you don't need the where if you write the OrderBy like that:
```suggestion
        var databasesToDelete = databases
            .OrderBy(db => 
                long.TryParse(
                    db.DatabaseName.DatabaseId.Replace("my-db-", "").Replace("my-restore-db-", ""),
                    long out creationDate) ?
                creationDate : long.MaxValue)
            .Take(5);    
```